### PR TITLE
adding merchant support

### PR DESCRIPTION
**Pull request by labhvam5**

### CODE REVIEW

File: connector/fyle_integrations_platform_connector/apis/merchants.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+from heapq import merge
```
Comment (Shwetabhk) [inline connector/fyle_integrations_platform_connector/apis/merchants.py:1]: is this used anywhere?

File: connector/fyle_integrations_platform_connector/apis/merchants.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+from traceback import print_tb
```
Comment (Shwetabhk) [inline connector/fyle_integrations_platform_connector/apis/merchants.py:2]: is this used anywhere?

File: connector/fyle_integrations_platform_connector/apis/merchants.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        """
+        params = {'order': 'updated_at.desc'}
+        params.update(self.query_params)
+
+        return params
+
+    def get(self):
+        generator = self.get_all_generator()
+        for items in generator:
+            merchant = items['data'][0]
```
Comment (Shwetabhk) [inline connector/fyle_integrations_platform_connector/apis/merchants.py:29]:
```suggestion
            merchants = items['data'][0]
```

File: connector/fyle_integrations_platform_connector/apis/merchants.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        
+        return merchant
+
+    def post(self, payload: List[str]):
+        """
+        Post data to Fyle 
+        """
+        generator = self.get_all_generator()
+        for items in generator:
+            merchant = items['data'][0]
```
Comment (Shwetabhk) [inline connector/fyle_integrations_platform_connector/apis/merchants.py:39]:
```suggestion
            merchants = items['data'][0]
```

File: connector/fyle_integrations_platform_connector/apis/merchants.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+                'field_name': 'Merchant',
+                'type': 'SELECT',
+                'options': merchant['options'],
+                'placeholder': merchant['placeholder'],
+                'category_ids': merchant['category_ids'],
+                'is_enabled': merchant['is_enabled'],
+                'is_custom': merchant['is_custom'],
+                'is_mandatory': merchant['is_mandatory'],
+                'code': merchant['code'],
+                'default_value': merchant['default_value'] if merchant['default_value'] else payload[0],
```
Comment (Shwetabhk) [inline connector/fyle_integrations_platform_connector/apis/merchants.py:52]: There is no need to add a default value
```suggestion
                'default_value': merchant['default_value'] if merchant['default_value'] else None,
```

Comment (labhvam5) [reply connector/fyle_integrations_platform_connector/apis/merchants.py:52]: This cannot be null if we are posting some options from our side. There will be a validation error.

**Review by Shwetabhk — APPROVED** [review]

**Pull request merged**